cmd/process-agent/main_common.go

```diff
@@ -109,18 +109,30 @@ func init() {
 func fixDeprecatedFlags() {
 	deprecatedFlags := []string{
 		// Global flags
-		"-config", "-ddconfig", "-sysprobe-config", "-pid", "-info", "-version", "-check",
+		"-config", "--config", "-sysprobe-config", "-pid", "-info", "-version", "-check",
 		// Windows flags
 		"-install-service", "-uninstall-service", "-start-service", "-stop-service", "-foreground",
 	}
 
+	replaceFlags := map[string]string{
+		"-config":  "--cfgpath",
+		"--config": "--cfgpath",
+	}
+
 	for i, arg := range os.Args {
 		for _, f := range deprecatedFlags {
 			if !strings.HasPrefix(arg, f) {
 				continue
 			}
-			fmt.Printf("WARNING: `%s` argument is deprecated and will be removed in a future version. Please use `-%[1]s` instead.\n", f)
-			os.Args[i] = "-" + os.Args[i]
+			replaceWith := replaceFlags[f]
+			if len(replaceWith) == 0 {
+				os.Args[i] = "-" + f
+				os.Args[i] = "-" + os.Args[i]
```

Comment:
Why are we prefixing the deprecated flags with `--`? Mind adding a comment about this in the code?
Also, could simplify to a single string concat with 
```suggestion
				os.Args[i] = "--" + f
```

Reply:
Unfortunately cannot do the concat because of `-config=foo` situation, example of this [use](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_container-process-agent.yaml#L9), I will add a comment though.